rust/src/transformations/make_stable_expr/expr_len/mod.rs

```diff
@@ -0,0 +1,93 @@
+use crate::core::{Function, MetricSpace, StabilityMap, Transformation};
+use crate::domains::{AtomDomain, ExprDomain, LogicalPlanDomain, MarginPub, SeriesDomain};
+use crate::error::*;
+use crate::metrics::{IntDistance, LpDistance, PartitionDistance};
+use crate::traits::{InfCast, InfMul, InfSqrt, ProductOrd};
+use crate::transformations::traits::UnboundedMetric;
+use polars_plan::dsl::{len, Expr};
+use polars_plan::logical_plan::LogicalPlan;
+
+#[cfg(test)]
+mod test;
+
+/// Polars operator to sum a column in a LazyFrame
+///
+/// | input_metric                              |
+/// | ----------------------------------------- |
+/// | `PartitionDistance<SymmetricDistance>`    |
+/// | `PartitionDistance<InsertDeleteDistance>` |
+///
+/// # Arguments
+/// * `input_domain` - ExprDomain
+/// * `input_metric` - valid selections shown in table above
+/// * `expr` - a length expression
+pub fn make_expr_len<MI, const P: usize>(
+    input_domain: ExprDomain,
+    input_metric: PartitionDistance<MI>,
+    expr: Expr,
+) -> Fallible<Transformation<ExprDomain, ExprDomain, PartitionDistance<MI>, LpDistance<P, f64>>>
+where
+    MI: 'static + UnboundedMetric,
+    (ExprDomain, PartitionDistance<MI>): MetricSpace,
+    (ExprDomain, LpDistance<P, f64>): MetricSpace,
+{
+    let Expr::Len = expr else {
+        return fallible!(MakeTransformation, "expected len expression");
+    };
+
+    // check that we are in a context where it is ok to break row-alignment
+    input_domain.context.break_alignment()?;
```

Comment:
I've seen this idiom before, but I'm not sure if we exercise it failing anywhere?